### PR TITLE
add dependabot for github actions

### PR DESCRIPTION
**Pull request by syphar**
seeing a3efa45f7f0fbc4ca7aefbd3042ae57351975e4a I thought this could be one case where we could use dependabot.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+version: 2
+updates:
+  - package-ecosystem: "github-actions"
+    directory: "/"
+    schedule:
+      interval: "daily"
```
Comment (jyn514) [inline .github/dependabot.yml:6]: Can we change this to only make PRs on major versions?
```suggestion
      interval: "daily"
    ignore:
      - dependency-name: "*"
        update-types: ["version-update:semver-patch", "version-update:semver-minor"]
```

**Comment by syphar** [comment]
@jyn514 having slept over it I'm not sure any more if this fits. 

- currently we're referencing actions only via major versions (`@v2`...). 
- which means we're automatically getting minor & patch updates. 

dependabot will actually _change_ how the steps are referenced to the full version (`@v2.1.1`...). 

Ignoring the patch/minor updates would lead to us _not getting minor or patch updates any more_, which is a change to what we had before. 

So the solutions are: 
1. use automatic PRs for the updates, but that means getting PRs for minor & patch updates if we want to keep these updates 
2. keep it as is, but remember doing the major updates manually

**Comment by jyn514** [comment]
> dependabot will actually change how the steps are referenced to the full version (@v2.1.1...).

Is there some way to tell it not to do this?

If not, I agree dependabot is not a good fit here :(

**Comment by syphar** [comment]
> > dependabot will actually change how the steps are referenced to the full version (@v2.1.1...).
> 
> Is there some way to tell it not to do this?

In the end it follows the same pattern we follow with the lockfile / freezing patch-versions. While explicitly reviewing these on PRs definitely makes sense in private projects, I lean towards loosening these checks for open source projects :) 



> 
> If not, I agree dependabot is not a good fit here :(

Yep, I'll close it.

**Pull request closed without merging**